File: lib/java/src/main/java/io/github/brenbar/uapi/internal/types/UFieldDeclaration.java
package io.github.brenbar.uapi.internal.types;

public class UFieldDeclaration {
    public final String fieldName;
    public final UTypeDeclaration typeDeclaration;
    public final boolean optional;

    public UFieldDeclaration(
            String fieldName,
            UTypeDeclaration typeDeclaration,
            boolean optional) {
        this.fieldName = fieldName;
        this.typeDeclaration = typeDeclaration;
        this.optional = optional;
    }
}

File: lib/ts/src/internal/types/UFieldDeclaration.ts
import { UTypeDeclaration } from 'uapi/internal/types/UTypeDeclaration';

export class UFieldDeclaration {
    fieldName: string;
    typeDeclaration: UTypeDeclaration;
    optional: boolean;

    constructor(fieldName: string, typeDeclaration: UTypeDeclaration, optional: boolean) {
        this.fieldName = fieldName;
        this.typeDeclaration = typeDeclaration;
        this.optional = optional;
    }
}
